docs/reference/style-guide.rst

```diff
@@ -0,0 +1,392 @@
+
+Fore the sake of consistency, please **do not** use the pattern:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - .. code:: reStructuredText
+       
+            Look, after this text starts a code block::
+       
+                echo "Hello, World!"
+     - Look, after this text starts a code block::
+       
+           echo "Hello, World!"
+
+Instead you can do this:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - .. code:: reStructuredText
+       
+            Look, after this text starts a code block:
+
+            .. code::
+       
+                echo "Hello, World!"
+     - Look, after this text starts a code block:
+
+       .. code::
+       
+           echo "Hello, World!"
+
+Definition lists
+----------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - .. code:: reStructuredText
+
+            Term 1:
+                Definition
+
+            Term 2:
+                Definition
+     - Term 1:
+           Definition
+       
+       Term 2:
+           Definition
+
+Cards
+-----
+
+.. grid:: 2
+    
+    .. grid-item::
+        :child-align: center
+
+        .. code:: reStructuredText
+
+            .. card:: Card title
+                Card content
+
+    .. grid-item::
+        :child-align: center
+
+        .. card:: Card title
+            
+            Card content
+
+Tabs
+----
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - .. code:: reStructuredText
+
+           .. tab-set::
+    
+               .. tab-item:: Label1
+                   :sync: key1
+       
+                   Content 1
+       
+               .. tab-item:: Label2
+                   :sync: key2
+       
+                    Content 2
+       
+           .. tab-set::
+       
+               .. tab-item:: Label1
+                   :sync: key1
+       
+                   Content 1
+       
+               .. tab-item:: Label2
+                   :sync: key2
+       
+                   Content 2
+     - .. tab-set::
+
+           .. tab-item:: Label1
+               :sync: key1
+   
+               Content 1
+   
+           .. tab-item:: Label2
+               :sync: key2
+   
+               Content 2
+   
+       .. tab-set::
+   
+           .. tab-item:: Label1
+               :sync: key1
+   
+               Content 1
+   
+           .. tab-item:: Label2
+               :sync: key2
+   
+               Content 2
+
+More useful markup
+------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - .. code:: reStructuredText
+
+           | Line 1
+           | Line 2
+           | Line 3
+     - | Line 1
+       | Line 2
+       | Line 3
+   * - .. code:: reStructuredText
+
+           ----
+     - A horizontal line
+     
+       (for technical reasons the horizontal line can not be displayed here)
+   * - .. code:: reStructuredText
+
+           .. This is a comment
+     - 
+
+
```

Comment:
We should probably include link formatting and options as well. First, we need to decide if we want to go ahead with translations, because that will have to factor into our decisions about how links should be treated (doing them the standard way as presented in the reST style guide that was the basis of this doc won't work very well and will make translators' jobs harder).